Add ConfigProvider

geerteltink opened the pull request:
Adds ConfigProvider with basic configuration.

geerteltink committed:
Added ConfigProvider with basic configuration

geerteltink committed:
Bring composer.lock insync with composer.json

michalbundyra commented on composer.json:
```diff
@@ -50,6 +50,9 @@
             "dev-release-3.0.0": "3.0.x-dev"
+        },
+        "zf": {
+            "config-provider": "Zend\\Expressive\\Router\\ConfigProvider"
```
As in the previous review, I would consider changing namespace for the repository.

michalbundyra commented on src/ConfigProvider.php:
```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-aurarouter for the canonical source repository
+ * @copyright Copyright (c) 2015-2018 Zend Technologies USA Inc. (https://www.zend.com)
```
Only 2018 please for new files.

webimpress commented on src/ConfigProvider.php:
```diff
@@ -0,0 +1,29 @@
+ * @copyright Copyright (c) 2015-2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-aurarouter/blob/master/LICENSE.md New BSD License
+ */
+
```
Remove empty line before strict type declaration.

michalbundyra commented on test/ConfigProviderTest.php:
```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-aurarouter for the canonical source repository
+ * @copyright Copyright (c) 2015-2018 Zend Technologies USA Inc. (https://www.zend.com)
```
2018 only please

michalbundyra commented on test/ConfigProviderTest.php:
```diff
@@ -0,0 +1,47 @@
+ * @copyright Copyright (c) 2015-2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-aurarouter/blob/master/LICENSE.md New BSD License
+ */
+
```
no new line before strict_type declaration

weierophinney replied:
Have we actually decided on this? The Stratigility repo has no space between the file-level docblock and the declaration, but the zend-expressive repo _does_ (a fact I was going to point out in your review of my latest patch there). We need to decide on a style and stick with one; right now, we have two in as many repos.

Personally, I prefer a line between them, as that makes it clear that the docblock is separate from the declaration.

geerteltink replied:
Not sure if we decide it here, but I prefer a new line in between as well.

michalbundyra replied:
@weierophinney @xtreamwayz I've jest checked, and I was sure that we shouldn't have the empty line there, but I'm really sorry. I was wrong! I prefer the line as well, and here is a previous comment by @weierophinney:
https://github.com/zendframework/zend-expressive-router/pull/41#discussion_r155356898

So lets keep the empty line, update other repositories to add this line, and ignore my all comments about it. Thanks.

michalbundyra commented on src/ConfigProvider.php:
```diff
@@ -0,0 +1,29 @@
+    public function getDependencies() : array
+    {
+        return [
+            'factories' => [
```
`invokables` probably...
https://github.com/zendframework/zend-expressive-skeleton/blob/release-3.0.0/src/ExpressiveInstaller/Resources/config/router-aura-router.php#L10

I would consider to have `invokables` for `AuraRouter` and `aliases` for `RouterInterface`

michalbundyra commented on test/ConfigProviderTest.php:
```diff
@@ -0,0 +1,47 @@
+    public function testInvocationReturnsArray() : array
+    {
+        $config = ($this->provider)();
+        self::assertInternalType('array', $config);
```
`$this->`... ?

michalbundyra commented on test/ConfigProviderTest.php:
```diff
@@ -0,0 +1,47 @@
+        self::assertInternalType('array', $config['dependencies']);
+        self::assertArrayHasKey('factories', $config['dependencies']);
+        self::assertInternalType('array', $config['dependencies']['factories']);
+        self::assertArrayHasKey(RouterInterface::class, $config['dependencies']['factories']);
```
`$this->` ... 5x ?

weierophinney commented:
@xtreamwayz Same comments from the fastroute package apply here:

- Use a subnamespace for the `ConfigProvider`.
- Use `$this->` instead of `self::` for now.

geerteltink commented:
> Use a subnamespace for the ConfigProvider.

@mwop Not sure what you mean by that. Change the namespace of the entire package to `Zend\Expressive\Aurarouter`? That would be a BC break for all router components. 
Change the name or move the ConfigProvider you create an inconsistency with other packages.

weierophinney commented:
> @mwop Not sure what you mean by that. Change the namespace of the entire package to Zend\Expressive\Aurarouter? That would be a BC break for all router components.
> Change the name or move the ConfigProvider you create an inconsistency with other packages.

I think you meant @weierophinney here. :smile:

No, I meant that you should add a subnamespace in _this_ package: `Zend\Expressive\Router\AuraRouter` (as the root namespace is `Zend\Expressive\Router`, and push the `ConfigProvider` under that: `Zend\Expressive\Router\AuraRouter\ConfigProvider`.

Otherwise, we'll have potential conflicts when trying to resolve the `Zend\Expressive\Router\ConfigProvider` class.

weierophinney committed:
Move ConfigProvider to subnamespace

Moves it to the `Zend\Expressive\Router\AuraRouter` subnamespace, to
prevent potential conflicts with other config providers under the
`Zend\Expressive\Router` namespace.

weierophinney committed:
Updates copyright range to reflect actual introduction

weierophinney committed:
Updates test assertion style

We use `$this->` everywhere else, essentially, so `self::` is
inconsistent. If we decided to go with the latter in the future, we can
do mass updates across all components.

weierophinney committed:
Adds CHANGELOG entry for #30

weierophinney commented:
Thanks, @xtreamwayz!
